[masterbots.ai] feat: integrate drizzle ORM

Bran18 opened the pull request:
## Summary by Sourcery

Integrate Drizzle ORM into the project to enhance database management capabilities. Implement new admin services and API for user management, including actions like blocking users and updating subscriptions. Update environment and docker-compose configurations to support the new database setup.

New Features:
- Integrate Drizzle ORM into the project, providing a new ORM layer for database interactions.
- Add admin service functions for user management, including blocking, unblocking, updating subscriptions, and setting free months.
- Introduce a new admin API for executing admin actions such as blocking users and updating subscriptions.

Enhancements:
- Add database configuration to the environment configuration file, supporting the new Drizzle ORM integration.

Deployment:
- Update docker-compose configuration to include a DATABASE_URL environment variable for database connectivity.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Release Notes

- **New Features**
	- Enhanced error handling and validation for admin actions.
	- Introduced an `adminApi` for executing administrative actions with type safety.
	- New functions for user management, including blocking/unblocking users and updating subscriptions.
	- Added a structured approach for handling administrative actions with schema validation.

- **Bug Fixes**
	- Improved robustness of API responses for various error scenarios.

- **Documentation**
	- Updated configuration for database connection and added new schema definitions for user management.

- **Chores**
	- Added environment variable for database connection in the Docker configuration.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Bran18 commented on packages/mb-drizzle/drizzle/meta/0001_snapshot.json:
```diff
@@ -0,0 +1,153 @@
+          "notNull": false
+        }
+      },
+      "indexes": {},
```
Will explore it

Bran18 commented on docker-compose.yml:
```diff
@@ -13,6 +13,8 @@ services:
       POSTGRES_USER: user
       POSTGRES_PASSWORD: pass
       POSTGRES_DB: masterbots
+      DATABASE_URL: "postgres://user:pass@localhost:5433/masterbots?sslmode=disable"
+
```
Agree, but i will wait for andler before it

AndlerRL replied:
@coderabbitai this is fine. The compose file is the local file to boot up our database locally. It can be either with localhost or the container name (I think this one is either "databases or "masterbots"). So it is not a risk because we are not exposing any production or remote development environment credentials.

AndlerRL commented on apps/masterbots.ai/services/admin/admin.service.ts:
```diff
@@ -0,0 +1,70 @@
+import type { AdminUserUpdate } from '@/types/types'
+
+export async function verifyAdmin() {
+  const session = await getServerSession()
```
We can include 3 sets of email domains:

- masterbots.ai 
- mbots.ai
- bitcash.org

cc - @Bran18

AndlerRL commented on apps/masterbots.ai/services/admin/admin.service.ts:
```diff
@@ -0,0 +1,70 @@
+import { getServerSession } from 'next-auth'
+import type { AdminUserUpdate } from '@/types/types'
+
+export async function verifyAdmin() {
+  const session = await getServerSession()
+  if (!session?.user?.email?.endsWith('@masterbots.ai')) {
+    throw new Error('Unauthorized: Admin access required')
+  }
+}
```
@coderabbitai do not worry, we already have a role level security with the "moderator" role: we read on the JWT the user permissions with the graphql API that we have. Each user has their respective role (users and moderators). So if they want to hack the email domain, they have to hack the JWT and see if the key matches with the DB and the credentials. So this is a verification in 2 places. Does that make sense?

AndlerRL commented on apps/masterbots.ai/services/admin/admin.service.ts:
```diff
@@ -0,0 +1,70 @@
+    where: eq(users.userId, userId)
+  })
+
+  if (user?.proUserSubscriptionId) {
```
@sourcery-ai please create an issue based on your comment regarding this issue. Yes, we need to be cautious on how we check the subscription free month. They can get up to 1 free month, no matter how many they refer to, we can offer another gift (like discounts) for those users but not another free month.

AndlerRL reviewed (approved):
Very good changes! I ran a review and added a new issue regarding a comment made by sourcery-ai, we can deal with the issue on another PR.

I left some feedback given by both AIs and resolved some of them while reviewing. After these updates, it will be good to go! :+1:

Bran18 commented:
> Very good changes! I ran a review and added a new issue regarding a comment made by sourcery-ai, we can deal with the issue on another PR.
> 
> I left some feedback given by both AIs and resolved some of them while reviewing. After these updates, it will be good to go! 👍

Awesome! I just added the recommendations from the AI and the one we talked about with the `admin-actions`
